File: project3-server client/Client/src/Frame.cpp
//
// Created by royek on 10/01/2020.
//

#include "Frame.h"

int Frame::frameId(0);

Frame::Frame(string stompCommand, map<string, string> &headers, string bodyFrame): stompCommand(stompCommand), headers(headers), frameBody(bodyFrame) {}

string Frame::getStompCommand() { return this->stompCommand; }
map<string, string>& Frame::getHeaders() { return  this->headers; }
string Frame::getFrameBody(){ return this->frameBody; }
int Frame::getFrameId() { return frameId++; }

File: project3-server client/Server/src/main/java/bgu/spl/net/srv/Frame.java
package bgu.spl.net.srv;

import java.util.Map;
import java.util.concurrent.atomic.AtomicInteger;

public class Frame {

    private static AtomicInteger frameId = new AtomicInteger(0);

    private String stompCommand;
    private Map<String, String> headers;
    private String frameBody;

    public Frame(String stompCommand, Map<String, String> headers, String frameBody)
    {
        this.stompCommand = stompCommand;
        this.headers = headers;
        this.frameBody = frameBody;
    }

    public String getStompCommand() { return this.stompCommand; }
    public Map<String, String> getHeaders() { return this.headers; }
    public String getFrameBody() { return this.frameBody; }
    public static Integer getFrameId() {
        frameId.incrementAndGet();
        return frameId.get();
    }

}
